src/containers/permissions/DataSetPermissionsContainer.js

```diff
@@ -0,0 +1,210 @@
+    (permissions.get(principal) || Map()).filter((ace :Ace, key :List<UUID>) => {
+      const dataSetId :UUID = key.get(0);
+      const atlasDataSet :?Map = atlasDataSets.get(dataSetId);
+      const entitySet :?EntitySet = entitySets.get(dataSetId);
+      const dataSetTitle :string = getDataSetField(atlasDataSet || entitySet, 'title');
+      return (
+        dataSetTitle.toLowerCase().includes(filterByQuery.toLowerCase())
+        && filterByPermissionTypes.every((pt :PermissionType) => ace?.permissions.includes(pt))
+      );
+    })
+  ), [
+    atlasDataSetsHash,
+    entitySetsHash,
+    filterByPermissionTypes,
+    filterByQuery,
+    permissionsHash,
+    principal,
+  ]);
+
+  const filteredPermissionsCount :number = filteredPermissions.count();
+  const pagePermissions :Map<List<UUID>, Ace> = filteredPermissions.slice(start, start + MAX_PER_PAGE);
+  const pageDataSetIds :Set<UUID> = pagePermissions.keySeq().flatten().toSet();
+  const pageDataSetIdsHash :number = pageDataSetIds.hashCode();
+  const pageAtlasDataSets :Map<UUID, Map> = useSelector(selectAtlasDataSets(pageDataSetIds));
+  const pageEntitySets :Map<UUID, EntitySet> = useSelector(selectEntitySets(pageDataSetIds));
+  const pageDataSets :Map<UUID, EntitySet | Map> = Map().merge(pageAtlasDataSets).merge(pageEntitySets);
+
+  useEffect(() => {
+    dispatch(initializeDataSetPermissions({ organizationId }));
+  }, [dispatch, organizationId]);
+
+  useEffect(() => {
+    if (!pageDataSetIds.isEmpty()) {
+      const pageAtlasDataSetIds = pageDataSetIds.filter((id :UUID) => atlasDataSetIds.has(id));
+      const pageEntitySetIds = pageDataSetIds.filter((id :UUID) => entitySetIds.has(id));
+      dispatch(
+        getPageDataSetPermissions({
+          atlasDataSetIds: pageAtlasDataSetIds,
+          entitySetIds: pageEntitySetIds,
+          organizationId,
+          withProperties: true,
+        })
+      );
+    }
+  }, [
+    atlasDataSetIds,
+    dispatch,
+    entitySetIds,
+    organizationId,
+    pageDataSetIdsHash,
+  ]);
+
+  const handleOnPageChange = (state :PaginationState) => {
+    paginationDispatch({
+      page: state.page,
+      start: state.start,
+      type: PAGE,
+    });
+    onSelect();
+  };
+
+  const toggleOpenDataSetCard = (dataSetId :UUID) => {
+    if (openDataSetId === dataSetId) {
+      setOpenDataSetId('');
+    }
+    else {
+      setOpenDataSetId(dataSetId);
+    }
+  };
+
+  const reducedRS :?RequestState = reduceRequestStates([getPermissionsRS, initializePermissionsRS]);
+
+  return (
+    <StackGrid gap={8}>
+      {
+        reducedRS === RequestStates.PENDING && (
+          <Spinner />
+        )
+      }
+      {
+        reducedRS === RequestStates.SUCCESS && filteredPermissionsCount === 0 && (
+          <Typography align="center">No datasets.</Typography>
+        )
+      }
+      {
+        reducedRS === RequestStates.SUCCESS && filteredPermissionsCount !== 0 && (
```

Comment:
is there a reason we aren't using `isPending` and `isSuccess` from `ReduxUtils` in `lattice-utils`?